### PR TITLE
Mostly golinting

### PR DESCRIPTION
**Pull request by kargakis**
@mfojtik fyi

### CODE REVIEW

**Comment by mfojtik** [comment]
@kargakis big kudos for doing this! LGTM @smarterclayton ?

**Comment by smarterclayton** [comment]
This is too large to review.  When doing changes like this, put all Godoc changes in their own commit and make it a separate pull.  Keep functional changes isolated.  Please break this pull up.

File: pkg/cmd/server/start/kube_connection_args.go [hunk 1]
```diff
@@ -24,19 +24,29 @@ type KubeConnectionArgs struct {
...
-	config := &KubeConnectionArgs{}
+	return &KubeConnectionArgs{
+		KubernetesAddr: flagtypes.Addr{
+			Value: "localhost:8443", 
+			DefaultScheme: "https", 
+			DefaultPort: 8443, 
+			AllowPrefix: true,
+		}.Default(),
 
-	config.KubernetesAddr = flagtypes.Addr{Value: "localhost:8443", DefaultScheme: "https", DefaultPort: 8443, AllowPrefix: true}.Default()
```
Comment (smarterclayton) [inline pkg/cmd/server/start/kube_connection_args.go]: This changes the behavior of this call in a breaking way - &config is no longer defined.

**Comment by kargakis** [comment]
Stashed everything else and kept just golint errors

**Comment by smarterclayton** [comment]
Approved LGTM [merge]

**Comment by kargakis** [comment]
Hmm, even though this PR is ok, @smarterclayton @danmcp you should do something about that approved merge. It re-runs with every new push...

**Comment by kargakis** [comment]
@mfojtik this is in, fyi

**Comment by kargakis** [comment]
> Hmm, even though this PR is ok, @smarterclayton @danmcp you should do something about that approved merge. It re-runs with every new push...

Nevermind

**Pull request merged**